papis/document.py

```diff
@@ -425,6 +425,14 @@ def get_notes(self) -> List[str]:
 
     def save(self) -> None:
         """Saves the current document fields into the info file."""
+
+        folder = self.get_main_folder()
+        if not folder or not os.path.exists(folder):
+            from papis.exceptions import DocumentFolderNotFound
+            logger.error("running doc.save() without first running doc.set_folder() "
+                         "will fail due to an unset info file path.")
```

Comment:
```suggestion
```
? The exception should have sufficient context about what happened and the caller can write a message if they want to.

Reply:
good point. I added context to the exception. I checked all the other places it is raised and I think that's a fair step. It took me some time to figure it out when I first encountered it.